File: extension/generate_icons.py
"""Generate default weather icons for the browser extension."""
from PIL import Image, ImageDraw
import os

SIZES = [16, 32, 48, 128]
ICONS_DIR = os.path.join(os.path.dirname(__file__), 'icons')

def generate_icon(size: int) -> Image.Image:
    """Generate a weather app icon with sun and cloud."""
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Background - rounded rectangle (blue sky)
    radius = int(size * 0.15)
    draw.rounded_rectangle(
        [0, 0, size - 1, size - 1],
        radius=radius,
        fill=(79, 195, 247)  # Light blue
    )
    
    # Sun (yellow circle)
    sun_radius = int(size * 0.2)
    sun_x = int(size * 0.3)
    sun_y = int(size * 0.3)
    draw.ellipse(
        [sun_x - sun_radius, sun_y - sun_radius, 
         sun_x + sun_radius, sun_y + sun_radius],
        fill=(255, 213, 79)  # Yellow
    )
    
    # Cloud (white overlapping circles)
    cloud_y = int(size * 0.55)
    cloud_x = int(size * 0.55)
    cloud_r = int(size * 0.12)
    
    # Draw cloud circles
    for offset_x, offset_y, r_mult in [
        (0, 0, 1.0),
        (cloud_r * 0.7, -cloud_r * 0.3, 0.9),
        (cloud_r * 1.4, 0, 0.85),
        (cloud_r * 0.4, cloud_r * 0.3, 0.7),
        (cloud_r * 1.0, cloud_r * 0.3, 0.7),
    ]:
        r = int(cloud_r * r_mult)
        x = int(cloud_x + offset_x)
        y = int(cloud_y + offset_y)
        draw.ellipse([x - r, y - r, x + r, y + r], fill=(255, 255, 255))
    
    return img

def main():
    os.makedirs(ICONS_DIR, exist_ok=True)
    
    for size in SIZES:
        icon = generate_icon(size)
        path = os.path.join(ICONS_DIR, f'default-{size}.png')
        icon.save(path, 'PNG')
        print(f'Generated: {path}')
    
    print('All icons generated successfully!')

if __name__ == '__main__':
    main()

File: extension/generate-icons.js
// Script to generate default extension icons
// Run with: node generate-icons.js

const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

const sizes = [16, 32, 48, 128];

function generateIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Background gradient (blue sky color)
  const gradient = ctx.createLinearGradient(0, 0, size, size);
  gradient.addColorStop(0, '#4FC3F7');
  gradient.addColorStop(1, '#0288D1');
  
  // Draw rounded rectangle
  const radius = size * 0.15;
  ctx.fillStyle = gradient;
  ctx.beginPath();
  ctx.roundRect(0, 0, size, size, radius);
  ctx.fill();
  
  // Draw sun
  ctx.fillStyle = '#FFD54F';
  const sunRadius = size * 0.25;
  const sunX = size * 0.35;
  const sunY = size * 0.35;
  ctx.beginPath();
  ctx.arc(sunX, sunY, sunRadius, 0, Math.PI * 2);
  ctx.fill();
  
  // Draw cloud
  ctx.fillStyle = '#FFFFFF';
  const cloudY = size * 0.55;
  const cloudX = size * 0.55;
  const cloudSize = size * 0.15;
  
  // Cloud circles
  ctx.beginPath();
  ctx.arc(cloudX, cloudY, cloudSize, 0, Math.PI * 2);
  ctx.arc(cloudX + cloudSize * 0.8, cloudY - cloudSize * 0.3, cloudSize * 0.8, 0, Math.PI * 2);
  ctx.arc(cloudX + cloudSize * 1.6, cloudY, cloudSize * 0.9, 0, Math.PI * 2);
  ctx.arc(cloudX + cloudSize * 0.4, cloudY + cloudSize * 0.3, cloudSize * 0.7, 0, Math.PI * 2);
  ctx.arc(cloudX + cloudSize * 1.2, cloudY + cloudSize * 0.3, cloudSize * 0.7, 0, Math.PI * 2);
  ctx.fill();
  
  return canvas.toBuffer('image/png');
}

// Create icons directory
const iconsDir = path.join(__dirname, 'icons');
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir);
}

// Generate icons
sizes.forEach(size => {
  const buffer = generateIcon(size);
  fs.writeFileSync(path.join(iconsDir, `default-${size}.png`), buffer);
  console.log(`Generated default-${size}.png`);
});

console.log('All icons generated successfully!');
